api/v1beta2/awsmachine_types.go

```diff
@@ -109,11 +109,21 @@ type AWSMachineSpec struct {
 	// +optional
 	AdditionalSecurityGroups []AWSResourceReference `json:"additionalSecurityGroups,omitempty"`
 
+	// VPC is a reference to the VPC to use when picking a subnet to use for this
+	// instance. Only valid if the subnet (id or filters) is also specified.
+	// +optional
+	VPC *AWSResourceReference `json:"vpc,omitempty"`
```

Comment:
Reading the related use case, I was wondering if we should be able to infer the VPC id from the subnet identifier? 

What could confuse a bit users with the changes proposed here is that we have `AdditionalSecurityGroups` and now `SecurityGroupOverrides`, which might conflict with each other?

Reply:
> Reading the related use case, I was wondering if we should be able to infer the VPC id from the subnet identifier?

I think this is possible, it would require rewriting the existing logic to always look up the VPC, but it does have the advantage of simplifying the fields for the AWSMachine. I'll take a look at doing this.

> What could confuse a bit users with the changes proposed here is that we have AdditionalSecurityGroups and now SecurityGroupOverrides, which might conflict with each other?

I agree there is some confusion because of the two fields for managing the same concept in AWS. The problem I tried to solve is that we need a different set of "default" SGs in the other VPC, because SGs are _bound_ to a particular VPC, and we can't re-use the cluster default SGs. So I provided a way for the user to supply their own. `AdditionalSecurityGroups` is insufficient because we need default SGs that are present in the secondary VPC. I don't care about `SecurityGroupOverrides` as a field, except as a way to provide SGs for the secondary VPC.

So the other option is we could eliminate the `SecurityGroupOverrides` field and instead have the CAPA duplicate the SGs from the cluster VPC into any secondary VPC, and use those for these nodes. But there would also be no good way to tell when we could tear down these SGs, without tracking them at the cluster level too. We'd probably have to add them to the AWSCluster or AWSManagedCluster's status field to keep track. Otherwise they might get left dangling after a cluster is torn down. Can you think of a better way to make this work?

Reply:
@vincepri updated to look up the VPC as needed, if you want to give it another look.